lxd/container_lxc.go

```diff
@@ -1728,7 +1728,11 @@ func (c *containerLXC) initLXC(config bool) error {
 	}
 
 	// Setup shmounts
-	err = lxcSetConfigItem(cc, "lxc.mount.entry", fmt.Sprintf("%s dev/.lxd-mounts none bind,create=dir 0 0", c.ShmountsPath()))
+	if lxc.HasApiExtension("mount_injection") || util.RuntimeLiblxcVersionAtLeast(3, 1, 0) {
```

Comment:
Do we still need the `||` here? Feels like anything that reports >= 3.1.0 would also report the API extension.